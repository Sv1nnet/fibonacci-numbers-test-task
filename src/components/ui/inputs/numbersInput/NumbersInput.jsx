import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-materialize';
import './numbers-input.scss';

/**
 *
 * @param {string} value - value that comes from target.value
 * @returns if value only contains digits or not
 */
export const isOnlyDigits = (value) => {
  const parsedValue = parseInt(value, 10);
  const onlyDigitals = /^\d+$/;

  return (!!value.match(onlyDigitals) && !Number.isNaN(parsedValue) && typeof parsedValue === 'number');
};

const NumbersInput = (props) => {
  const [inputValue, setInputValue] = useState('');
  const { disabled, onKeyUp } = props;

  const onChange = (e) => {
    const { target } = e;

    if (isOnlyDigits(target.value) || target.value === '') {
      setInputValue(target.value);
      props.onChange(e);
    }
  };

  return (
    <div className="container numbers-input-container">
      <TextInput
        className="center-align numbers-input"
        placeholder="Input a number"
        value={inputValue}
        onChange={onChange}
        onKeyUp={onKeyUp}
        disabled={disabled}
      />
    </div>
  );
};

NumbersInput.propTypes = {
  onKeyUp: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

NumbersInput.defaultProps = {
  disabled: false,
};

export default NumbersInput;
