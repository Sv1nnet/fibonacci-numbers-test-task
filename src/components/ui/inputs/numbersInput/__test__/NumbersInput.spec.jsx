import React from 'react';
import {
  jest,
  it,
  describe,
  expect,
} from '@jest/globals';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'materialize-css';
import toJson from 'enzyme-to-json';
import NumbersInput from '../NumbersInput';

configure({ adapter: new Adapter() });

describe('Testing NumbersInput component', () => {
  it('Should match snapshot with empty props', () => {
    const fn = jest.fn();
    const component = shallow(<NumbersInput onKeyUp={fn} onChange={fn} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('NumbersInput should have input with empty value', () => {
    const fn = jest.fn();
    const component = mount(<NumbersInput onKeyUp={fn} onChange={fn} />);
    const input = component.find('input');

    input.simulate('change', { target: { value: 'a' } });
    expect(input.instance().value).toBeFalsy();
  });

  it('NumbersInput should have input with number value', () => {
    const value = '5';
    const fn = jest.fn();
    const component = mount(<NumbersInput onKeyUp={fn} onChange={fn} />);
    const input = component.find('input');

    input.simulate('change', { target: { value } });
    expect(input.instance().value).toBe(value);
  });

  it('NumbersInput should have input with number value', () => {
    const value = '5';
    const fn = jest.fn();
    const component = mount(<NumbersInput onKeyUp={fn} onChange={fn} />);
    const input = component.find('input');

    input.simulate('change', { target: { value } });
    expect(input.instance().value).toBe(value);
  });

  it('NumbersInput should be disabled it "disable" prop provided as true', () => {
    const fn = jest.fn();
    const component = mount(<NumbersInput disabled onKeyUp={fn} onChange={fn} />);
    const input = component.find('input');

    expect(input.instance().disabled).toBe(true);
  });
});
