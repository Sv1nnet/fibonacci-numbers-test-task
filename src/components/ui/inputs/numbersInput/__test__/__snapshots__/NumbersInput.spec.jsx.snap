// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NumbersInput component Should match snapshot with empty props 1`] = `
<div
  className="container numbers-input-container"
>
  <TextInput
    className="center-align numbers-input"
    disabled={false}
    id="TextInput-3"
    onChange={[Function]}
    onKeyUp={[MockFunction]}
    placeholder="Input a number"
    value=""
  />
</div>
`;
